xwiki-commons-core/xwiki-commons-jakartabridge/xwiki-commons-jakartabridge-servlet/src/main/java/org/xwiki/jakartabridge/servlet/JakartaServletBridge.java

```diff
@@ -0,0 +1,1002 @@
+    public static Collection<Part> toJakarta(Collection<javax.servlet.http.Part> jakarta)
+    {
+        return JakartaBridge.toJakarta(jakarta, JavaxToJakartaPart::new, JakartaToJavaxPart::new);
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Enumeration<javax.servlet.Servlet> toJavax(Enumeration<Servlet> jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return Collections.enumeration(Collections.list(jakarta).stream().map(JakartaServletBridge::toJavax).toList());
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static Enumeration<Servlet> toJakarta(Enumeration<javax.servlet.Servlet> javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        return Collections.enumeration(Collections.list(javax).stream().map(JakartaServletBridge::toJakarta).toList());
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static javax.servlet.http.Cookie[] toJavax(Cookie[] jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        javax.servlet.http.Cookie[] javax = new javax.servlet.http.Cookie[jakarta.length];
+
+        for (int i = 0; i < jakarta.length; ++i) {
+            javax[i] = toJavax(jakarta[i]);
+        }
+
+        return javax;
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static Cookie[] toJakarta(javax.servlet.http.Cookie[] javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        Cookie[] jakarta = new Cookie[javax.length];
+
+        for (int i = 0; i < javax.length; ++i) {
+            jakarta[i] = toJakarta(javax[i]);
+        }
+
+        return jakarta;
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static EnumSet<javax.servlet.DispatcherType> toJavax(EnumSet<DispatcherType> jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return EnumSet.copyOf(jakarta.stream().map(JakartaServletBridge::toJavax).collect(Collectors.toList()));
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static EnumSet<DispatcherType> toJakarta(EnumSet<javax.servlet.DispatcherType> javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        return EnumSet.copyOf(javax.stream().map(JakartaServletBridge::toJakarta).collect(Collectors.toList()));
+    }
+
+    //////////////////////////////////////////////////
+    // Unsupported
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static javax.servlet.descriptor.JspConfigDescriptor toJavax(JspConfigDescriptor jakarta)
+    {
+        // TODO
```

Comment:
You still plan to implement these methods?

Reply:
Not immediately, it did not seem to worth the trouble given that we have not using those concepts in XWiki. But eventually, probably.

Reply:
It just seems strange to me that some random methods are missing without any indication in any documentation. I find this dangerous as it could cause random failures at runtime. You're certain that those methods that are missing won't be needed by some code that we don't control? I haven't fully checked where exactly those wrappers are used. In any case, if there is any risk, we should make sure that we don't use these wrappers in XS itself.

Reply:
> In any case, if there is any risk, we should make sure that we don't use these wrappers in XS itself.

Unless I don't understand what you mean here, I already answered that: "we are not using those concepts in XWiki". Each of those methods is a rabbit hole of new wrappers to add.